File: app/src/main/java/com/antyzero/smoksmog/logger/AndroidLogger.java
package com.antyzero.smoksmog.logger;

import android.util.Log;

/**
 * Typical Android SDK loggining
 */
public class AndroidLogger implements Logger {

    @Override
    public void v( String tag, String message ) {
        Log.v( tag, message );
    }

    @Override
    public void v( String tag, String message, Throwable throwable ) {
        Log.v( tag, message, throwable );
    }

    @Override
    public void d( String tag, String message ) {
        Log.d( tag, message );
    }

    @Override
    public void d( String tag, String message, Throwable throwable ) {
        Log.d( tag, message, throwable );
    }

    @Override
    public void i( String tag, String message ) {
        Log.i( tag, message );
    }

    @Override
    public void i( String tag, String message, Throwable throwable ) {
        Log.i( tag, message, throwable );
    }

    @Override
    public void w( String tag, String message ) {
        Log.w( tag, message );
    }

    @Override
    public void w( String tag, String message, Throwable throwable ) {
        Log.w( tag, message, throwable );
    }

    @Override
    public void e( String tag, String message ) {
        Log.v( tag, message );
    }

    @Override
    public void e( String tag, String message, Throwable throwable ) {
        Log.v( tag, message, throwable );
    }


}

File: android/src/main/kotlin/smoksmog/logger/AndroidLogger.kt
package smoksmog.logger

import android.util.Log

/**
 * Typical Android SDK logging
 */
class AndroidLogger : Logger {

    override fun v(tag: String, message: String) {
        Log.v(tag, message)
    }

    override fun v(tag: String, message: String, throwable: Throwable) {
        Log.v(tag, message, throwable)
    }

    override fun d(tag: String, message: String) {
        Log.d(tag, message)
    }

    override fun d(tag: String, message: String, throwable: Throwable) {
        Log.d(tag, message, throwable)
    }

    override fun i(tag: String, message: String) {
        Log.i(tag, message)
    }

    override fun i(tag: String, message: String, throwable: Throwable) {
        Log.i(tag, message, throwable)
    }

    override fun w(tag: String, message: String) {
        Log.w(tag, message)
    }

    override fun w(tag: String, message: String, throwable: Throwable) {
        Log.w(tag, message, throwable)
    }

    override fun e(tag: String, message: String) {
        Log.v(tag, message)
    }

    override fun e(tag: String, message: String, throwable: Throwable) {
        Log.v(tag, message, throwable)
    }


}
